src/mam4xx/modal_aer_opt.hpp

```diff
@@ -0,0 +1,1665 @@
+#ifndef MAM4XX_MODAL_AER_OPT_HPP
+#define MAM4XX_MODAL_AER_OPT_HPP
+
+#include <Kokkos_Complex.hpp>
+#include <haero/math.hpp>
+#include <mam4xx/aero_config.hpp>
+#include <mam4xx/modal_aero_calcsize.hpp>
+#include <mam4xx/ndrop.hpp>
+#include <mam4xx/water_uptake.hpp>
+
+namespace mam4 {
+namespace modal_aer_opt {
+
+using View1D = DeviceType::view_1d<Real>;
+using View2D = DeviceType::view_2d<Real>;
+using View3D = DeviceType::view_3d<Real>;
+using ComplexView2D = DeviceType::view_2d<Kokkos::complex<Real>>;
+using ComplexView1D = DeviceType::view_1d<Kokkos::complex<Real>>;
+using View5D = Kokkos::View<Real *****>;
+using View0D = Kokkos::View<Real>;
+
+using ConstColumnView = haero::ConstColumnView;
+
+constexpr int pver = mam4::nlev;
+constexpr int ntot_amode = mam4::AeroConfig::num_modes();
+// FIXME:  is top_lev equal to 1 in aerosol optics ?
+constexpr int top_lev = 0;
+//
+constexpr int pcnst = 40;
+// ! min, max aerosol surface mode radius treated [m]
+constexpr Real rmmin = 0.01e-6;
```

Comment:
Arguably these are `BAD_CONSTANT`s

Reply:
But maybe not, if they are mode size boundaries, in which case they are model parameters, I guess.

Reply:
I added `BAD CONSTANT`